### PR TITLE
Feature/display agent info

### PR DESCRIPTION
**Pull request by asodemann18**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- Agent side has no layout/info

## What is the new behavior?

- Agent side layout is built out
- Agent can see pending trips
- Agent can approve/delete pending trips
- Agent can see YTD revenue
- Agent can see travelers who are currently on trips for the current day
- Button created for searching all travelers
- Travelers can now see estimated cost 
## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- Also refactored code based on suggestions from previous PRs
- Reviewer can just focus on the index and domUpdates files

### CODE REVIEW

File: src/AllTravelers.js [hunk 1]
```diff
@@ -9,41 +9,38 @@ class AllTravelers {
 
   getTrips(theStatus) {
```
Comment (robertguturing) [inline src/AllTravelers.js:10]: Thanks for your hard work in refactoring this method! Looks a lot better

File: src/AllTravelers.js [hunk 2]
```diff
@@ -9,41 +9,38 @@ class AllTravelers {
...
+      tripsDetails.image = destination.image;
+      tripsDetails.alt = destination.alt;
+      tripsDetails.status = trip.status;
+      return tripsDetails;
+    })
+  }
+
   calculateTravelerCost() {
     return this.trips.filter(trip => trip.date <= moment().format('YYYY/MM/DD') && trip.status === 'approved').map(trip => {
+      const destination = this.destinations.find(dest => dest.id === trip.destinationID)
```
Comment (robertguturing) [inline src/AllTravelers.js:38]: Thanks for making this refactor as well!

File: src/domUpdates.js [hunk 3]
```diff
@@ -22,24 +22,50 @@ const domUpdates = {
...
     return welcome.innerHTML = `Welcome ${theTraveler.user.name.split(" ")[0]}!`;
   },
 
-  displayTravelersTrips(theTraveler, theStatus) {
-    const tripInfoSection = document.querySelector('.trip-info');
-    const tripDetails = theTraveler.getTrips(theStatus).sort((a,b) => moment(b.date) - moment(a.date));
-    const formattedTripDetails = tripDetails.map(detail => this.getTripFormat(detail)).join('');
-    tripInfoSection.innerHTML = formattedTripDetails;    
+  displayTrips(location, travelerData, status, format) {
+    const section = document.querySelector(location);
```
Comment (attom2) [inline src/domUpdates.js:31]: section might not be the most semantic, what type of section is it?

File: src/index.js [hunk 4]
```diff
@@ -27,36 +27,86 @@ const fetchData = () => {
...
       traveler = new Traveler(2, dataSet.travelersData, dataSet.tripsData, dataSet.destinationsData);
-      domUpdates.displayWelcome(traveler);
-      domUpdates.displayTravelersTrips(traveler);
-      tripDestinations.addEventListener('click', domUpdates.displayDestinationList(allTravelers));     
-      submitBtn.addEventListener('click', function() {
-        // domUpdates.displayNewTripCost(traveler);
-        postTrip(allTravelers, traveler);
-      });
-      domUpdates.displayTravelerCosts(traveler);
+      travelerPage(traveler, allTravelers);
```
Comment (attom2) [inline src/index.js:34]: this looks a lot cleaner!

File: src/index.js [hunk 5]
```diff
@@ -27,36 +27,86 @@ const fetchData = () => {
...
+    domUpdates.displayWelcome(traveler);
+    domUpdates.displayTrips('.trip-info', traveler, "", 'getTripFormat');
+    tripDestinations.addEventListener('click', domUpdates.displayDestinationList(allTravelers));     
+    submitBtn.addEventListener('click', function() {
+      postTrip(allTravelers, traveler);
+    });
+    costBtn.addEventListener('click', function() {
+      domUpdates.displayNewTripCost(traveler);
+      document.addEventListener('click', function(event) {
+        const eventParent = event.target.parentNode.parentNode.parentNode.parentNode.parentNode.classList.contains('overlay')
```
Comment (attom2) [inline src/index.js:61]: This is a lot of parentNodes!

Comment (asodemann18) [reply src/index.js:61]: yeah....I don't want to talk about how long I spent trying to not go that route :(

**Pull request merged**